### DISCUSSION TITLE
Is it possible to serve the generated sitemap over SSR?

### DISCUSSION THREAD

**Discussion by stewones** [analogjs/analog#1183] [Q&A]
I'm about to launch a big site built with analog, but just realized I'm unable to serve the sitemap 💀

this is my `vite.config.ts`

```ts
      analog({
        prerender: {
          routes: [...prerenderContentRoutes(contentRoutes)],
          sitemap: {
            host: process.env['VITE_ANALOG_PUBLIC_BASE_URL'] || '',
          },
        },
        nitro: {
          dev: mode !== 'production',
          // tried this 👇 but no luck
          // routeRules: {
          //   '/sitemap.xml': {
          //     static: true,
          //   },
          // },
        },
      }),
```

then if I build with `BUILD_PRESET=bun` and execute the server `bun analog/server/index.mjs`

it generates the sitemap file under `analog/public`,  but the route `/sitemap.xml` doesn't work. everything else runs well though.

am I missing something? 🤔

**Comment by stewones** [comment]
@brandonroberts FWIW I took this example of you and tried to setup sitemap there. got the same issue, it returns 404 for `/sitemap.xml` when running the SSRed version

https://github.com/brandonroberts/analog-angular-ssr

```ts
  plugins: [
    analog({
      prerender: {
        routes: ['/'],
        sitemap: {
          host: process.env['VITE_ANALOG_PUBLIC_BASE_URL'] || '',
        },
      },
    }),
  ],
  ```

**Reply by stewones** [reply]
I think the problem is because the sitemap is being generated too late. The current config in `vite-plugin-nitro` expects assets to be present in the `dist/client` folder.

```sh
dist/client/assets/analogjs-content-BPYoW9wI.js  138.84 kB │ gzip: 44.94 kB
dist/client/assets/index-OYq4_xgD.js             269.66 kB │ gzip: 85.13 kB
✓ built in 2.88s
Building SSR application...
{
  baseName: "public",
  dir: "/Users/stewan/dev/analog-bun-ssr-example-main/dist/client",
}
{
  baseName: "public",
  dir: "/Users/stewan/dev/analog-bun-ssr-example-main/dist/client",
}
vite v5.2.6 building SSR bundle for production...
✓ 31 modules transformed.
dist/ssr/assets/index.page-DbBZRfie.js            2.49 kB
dist/ssr/assets/analogjs-content-Bd5qOT4v.js     19.03 kB
dist/ssr/assets/xhr2-CDOpgsGh.js                 36.45 kB
dist/ssr/main.server.js                       1,364.64 kB
✓ built in 3.12s
```

but when the server is building there's no sitemap there yet as the sitemap is the last item in the pipeline.

```sh
Prerendering static pages...
Building Server...
Building Sitemap...
Writing sitemap at /Users/stewan/dev/analog-bun-ssr-example-main/dist/analog/public/sitemap.xml
```

so my workaround for now is to build the sitemap myself and colocate it in the `src/assets` folder before anything.

**Reply by brandonroberts** [reply]
That's correct. The sitemap is built last, but maybe we could build it before building the SSR server so its included in the client assets.

**Reply by brandonroberts** [reply]
PR here to make the sitemap available through SSR

https://github.com/analogjs/analog/pull/1201
